### PR TITLE
gcoap: remove deprecated function gcoap_req_send()

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This function was deprecated [1,5 years ago](https://github.com/RIOT-OS/RIOT/commit/af1eca90#diff-d696c4c71e4cef0f608c90caebbabf0bR357), so I think we can remove it now. Additionally I renamed `gcoap_req_send2()` to `gcoap_req_send()` and provided a migration wrapper around it named `gcoap_req_send2()` that is already deprecated with removal slated for the 2019.10 release.

### Testing procedure
Compiling `examples/gcoap` should still work when all commits are used.

### Issues/PRs references
None

### CODE REVIEW

**Comment by miri64** [comment]
> A few other places need to be updated. Look for "gcoap_req_send2" in the repository. This includes the module documentation, uses by cord_ep and cord_ep_sim, and the gcoap example.

Agreed, but this I would prefer to do in follow-ups. This is exactly one of the reasons why I introduced the  legacy wrapper ;-).

**Comment by miri64** [comment]
> Finally, if we have not developed a deprecation window policy, I would be inclined to keep the deprecated gcoap_req_send2() until the 2020.01 release. I expect almost everyone that uses gcoap also uses this function. So since there are 3 gcoap users, then probably 2 of them use it. I wouldn't want to make both of them mad. :-)

Okay.

File: sys/include/net/gcoap.h [hunk 1]
```diff
@@ -592,26 +592,30 @@ static inline ssize_t gcoap_request(coap_pkt_t *pdu, uint8_t *buf, size_t len,
...
- * @param[in] port          Port at the destination
+ * @param[in] remote        Destination for the packet
  * @param[in] resp_handler  Callback when response received, may be NULL
  *
  * @return  length of the packet
  * @return  0 if cannot send
  */
-size_t gcoap_req_send(const uint8_t *buf, size_t len, const ipv6_addr_t *addr,
-                      uint16_t port, gcoap_resp_handler_t resp_handler);
+size_t gcoap_req_send2(const uint8_t *buf, size_t len,
```
[the remark this answers is not in the document]
Comment (miri64) [reply sys/include/net/gcoap.h]: Yepp. Will fix

**Comment by kb2ma** [comment]
Looks good, please squash.

**Comment by miri64** [comment]
Squashed.

**Pull request merged**